### PR TITLE
nmh: init at 1.7.1

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

These tools are mainly useful for writing scripts that manipulate `claws-mail`'s mail folders.  I doubt anybody uses `nmh` to read their email anymore, and most other mail systems have migrated to `qmail`'s `maildir` format.

Personally, I use mostly `notmuch`, but still occasionally launch `claws-mail`, so I keep my mail in `notmuch`-indexed `nmh` folders.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:62]:
```suggestion
    description = "New MH Mail Handling System";
```

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:68]:
```suggestion
    longDescription = ''
```

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:75]:
```suggestion
```

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:88]:
```suggestion

      These tools are mainly useful for writing scripts that manipulating
      claws-mail's mail folders.  Most other mail clients have migrated to
      maildir.
    '';
```

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:28]:
```suggestion
  postPatch = ''
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by amjoseph-nixpkgs** [comment]
@ofborg eval

**Comment by amjoseph-nixpkgs** [comment]
@ofborg eval

Darwin builders are stuck.

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:21]:
```suggestion
  pname = "nmh";
  version = "1.7.1";

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by amjoseph-nixpkgs** [comment]
@ofborg eval

**Comment by amjoseph-nixpkgs** [comment]
@AndersonTorres do you have any further requests for this PR?

File: pkgs/tools/networking/nmh/default.nix
Comment (AndersonTorres) [inline pkgs/tools/networking/nmh/default.nix:19]:
```suggestion
```

**Review by AndersonTorres — APPROVED** [review]

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build nmh

**Comment by amjoseph-nixpkgs** [comment]
Rebased.

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build nmh

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build nmh

**Comment by amjoseph-nixpkgs** [comment]
The builds work fine on ofborg's aarch64-linux and my x86_64-linux, but not ofborg's x86_64-linux.  I'm going to merge this (it has an approval) and if I was wrong I'll deal with it during ZHF.

**Pull request merged**
